src/modal/alert-modal.interface.ts

```diff
@@ -46,8 +46,9 @@ export interface AlertModalData {
 	buttons?: Array<ModalButton>;
 	/**
 	 * Callback for non-specific close events. `return false;` to prevent the modal from closing
+	 * Property trigger has a value of `overlay` or `close`
 	 */
-	close?: Function;
+	close?: (trigger: "overlay" | "close") => boolean;
```

Comment:
Totally my bad for not catching this @HYACCCINT, and thanks @zvonimirfras for indirectly pointing it out ... this should be:
```suggestion
	close?: (trigger?: "overlay" | "close") => boolean;
```

so that consumers aren't required to accept the `trigger` argument

Reply:
oops I definitely should have noticed this. sorry!